src/main/java/com/example/Main.java

```diff
@@ -24,7 +24,30 @@ public static HttpServer startServer() {
         // create a resource config that scans for JAX-RS resources and providers
         // in com.example package
         final ResourceConfig rc = new ResourceConfig().packages("com.example");
-
+        
+        
+        
+        // querying the database for a customer list
+        String query = "SELECT Name, PostalAddress, ZipCode, BankAccountNo " +
+        "FROM Customer " +
+        "WHERE CustomerId =" + CustomerId;
+        
+        ResultSet rs = executeQuery(query)
+        
+        
+        
+        // checks if the zip code entered matches the regular expression
+        zipCodeRegex = "^\d{5}(?:[-\s]\d{4})?$"
+        
+        if !enteredZipCode.matches(zipCodeRegex) {
+            throw new InvalidZipCodeException
+        }
+        
```

Comment:
I believe you should write a database migration to deal with the rows in the database that currently have a ZIP code that doesn't match the regular expression. 

It is desirable to avoid having inconsistencies in the database.